dsl/pace/dsl/stencil.py

```diff
@@ -804,14 +795,20 @@ def get_quantity_halo_spec(
 class StencilFactory:
     """Configurable class which creates stencil objects."""
 
-    def __init__(self, config: StencilConfig, grid_indexing: GridIndexing):
+    def __init__(self, config: StencilConfig, grid_indexing: GridIndexing, comm: Comm):
         """
         Args:
             config: gt4py-specific stencil configuration
             grid_indexing: configuration for domain and halo indexing`
         """
         self.config: StencilConfig = config
         self.grid_indexing: GridIndexing = grid_indexing
+        self.comm: Comm = comm
+
+        # Replace {rank} in self.config.cache_dir with the communicator's rank
+        self.config.cache_dir = self.config.cache_dir.replace(
+            "{rank}", str(comm.Get_rank())
+        )
```

Comment:
Is the idea here to allow 6 ranks to use one single cache as well? unless specified by the namelist to have 6 separate caches?

Reply:
Yeah, the user can basically select programmatically, or through the driver yaml config.